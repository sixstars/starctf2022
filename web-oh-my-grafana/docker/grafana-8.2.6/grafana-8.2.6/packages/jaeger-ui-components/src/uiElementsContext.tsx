// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React, { ReactElement } from 'react';

export type TooltipPlacement =
  | 'top'
  | 'left'
  | 'right'
  | 'bottom'
  | 'topLeft'
  | 'topRight'
  | 'bottomLeft'
  | 'bottomRight'
  | 'leftTop'
  | 'leftBottom'
  | 'rightTop'
  | 'rightBottom';
export type PopoverProps = {
  children: ReactElement;
  content: ReactElement;
  arrowPointAtCenter?: boolean;
  overlayClassName?: string;
  placement?: TooltipPlacement;
};

export const UIPopover: React.ComponentType<PopoverProps> = function UIPopover(props: PopoverProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Popover {...props} />;
      }}
    </GetElementsContext>
  );
};

export type TooltipProps = {
  title: string | ReactElement;
  getPopupContainer?: (triggerNode: Element) => HTMLElement;
  overlayClassName?: string;
  children: ReactElement;
  placement?: TooltipPlacement;
  mouseLeaveDelay?: number;
  arrowPointAtCenter?: boolean;
  onVisibleChange?: (visible: boolean) => void;
};

export const UITooltip: React.ComponentType<TooltipProps> = function UITooltip(props: TooltipProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Tooltip {...props} />;
      }}
    </GetElementsContext>
  );
};

export type IconProps = {
  type: string;
  className?: string;
  onClick?: React.MouseEventHandler<any>;
};

export const UIIcon: React.ComponentType<IconProps> = function UIIcon(props: IconProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Icon {...props} />;
      }}
    </GetElementsContext>
  );
};

export type DropdownProps = {
  overlay: React.ReactNode;
  placement?: 'topLeft' | 'topCenter' | 'topRight' | 'bottomLeft' | 'bottomCenter' | 'bottomRight';
  trigger?: Array<'click' | 'hover' | 'contextMenu'>;
  children?: React.ReactNode;
};

export const UIDropdown = function UIDropdown(props: DropdownProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Dropdown {...props} />;
      }}
    </GetElementsContext>
  );
};

export type MenuProps = {
  children?: React.ReactNode;
};

export const UIMenu = function UIMenu(props: MenuProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Menu {...props} />;
      }}
    </GetElementsContext>
  );
};

export type MenuItemProps = {
  children?: React.ReactNode;
};

export const UIMenuItem = function UIMenuItem(props: MenuItemProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.MenuItem {...props} />;
      }}
    </GetElementsContext>
  );
};

export type ButtonHTMLType = 'submit' | 'button' | 'reset';
export type ButtonProps = {
  children?: React.ReactNode;
  className?: string;
  htmlType?: ButtonHTMLType;
  icon?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
};

export const UIButton = function UIButton(props: ButtonProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Button {...props} />;
      }}
    </GetElementsContext>
  );
};

export type DividerProps = {
  className?: string;
  type?: 'vertical' | 'horizontal';
};

export const UIDivider = function UIDivider(props: DividerProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Divider {...props} />;
      }}
    </GetElementsContext>
  );
};

export type InputProps = {
  autosize?: boolean | null;
  placeholder?: string;
  onChange: (value: React.ChangeEvent<HTMLInputElement>) => void;
  suffix: React.ReactNode;
  value?: string;
};

export const UIInput: React.FC<InputProps> = function UIInput(props: InputProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.Input {...props} />;
      }}
    </GetElementsContext>
  );
};

export type InputGroupProps = {
  className?: string;
  compact?: boolean;
  style?: React.CSSProperties;
  children?: React.ReactNode;
};

export const UIInputGroup = function UIInputGroup(props: InputGroupProps) {
  return (
    <GetElementsContext>
      {(elements: Elements) => {
        return <elements.InputGroup {...props} />;
      }}
    </GetElementsContext>
  );
};

export type Elements = {
  Popover: React.ComponentType<PopoverProps>;
  Tooltip: React.ComponentType<TooltipProps>;
  Icon: React.ComponentType<IconProps>;
  Dropdown: React.ComponentType<DropdownProps>;
  Menu: React.ComponentType<MenuProps>;
  MenuItem: React.ComponentType<MenuItemProps>;
  Button: React.ComponentType<ButtonProps>;
  Divider: React.ComponentType<DividerProps>;
  Input: React.ComponentType<InputProps>;
  InputGroup: React.ComponentType<InputGroupProps>;
};

/**
 * Allows for injecting custom UI elements that will be used. Mainly for styling and removing dependency on
 * any specific UI library but can also inject specific behaviour.
 */
const UIElementsContext = React.createContext<Elements | undefined>(undefined);
UIElementsContext.displayName = 'UIElementsContext';
export default UIElementsContext;

type GetElementsContextProps = {
  children: (elements: Elements) => React.ReactNode;
};

/**
 * Convenience render prop style component to handle error state when elements are not defined.
 */
export function GetElementsContext(props: GetElementsContextProps) {
  return (
    <UIElementsContext.Consumer>
      {(value: Elements | undefined) => {
        if (!value) {
          throw new Error('Elements context is required. You probably forget to use UIElementsContext.Provider');
        }
        return props.children(value);
      }}
    </UIElementsContext.Consumer>
  );
}
