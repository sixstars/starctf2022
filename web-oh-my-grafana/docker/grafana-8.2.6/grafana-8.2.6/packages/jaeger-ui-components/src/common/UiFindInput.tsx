// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';

import { TNil } from '../types/index';
import { UIIcon, UIInput } from '../uiElementsContext';

type Props = {
  allowClear?: boolean;
  inputProps: Record<string, any>;
  location: Location;
  match: any;
  trackFindFunction?: (str: string | TNil) => void;
  value: string | undefined;
  onChange: (value: string) => void;
};

export default class UiFindInput extends React.PureComponent<Props> {
  static defaultProps: Partial<Props> = {
    inputProps: {},
    trackFindFunction: undefined,
    value: undefined,
  };

  clearUiFind = () => {
    this.props.onChange('');
  };

  render() {
    const { allowClear, inputProps, value } = this.props;

    const suffix = (
      <>
        {allowClear && value && value.length && <UIIcon type="close" onClick={this.clearUiFind} />}
        {inputProps.suffix}
      </>
    );

    return (
      <UIInput
        autosize={null}
        placeholder="Find..."
        {...inputProps}
        onChange={(e) => this.props.onChange(e.target.value)}
        suffix={suffix}
        value={value}
      />
    );
  }
}
