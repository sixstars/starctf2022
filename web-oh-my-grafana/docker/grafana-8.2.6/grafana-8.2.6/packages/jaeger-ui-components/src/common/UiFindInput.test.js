// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { shallow } from 'enzyme';
// eslint-disable-next-line lodash/import-scope
import _ from 'lodash';

import UiFindInput from './UiFindInput';
import { UIInput } from '../uiElementsContext';

const debounceMock = jest.spyOn(_, 'debounce').mockImplementation((func) => {
  return Object.assign(func, { cancel: jest.fn(), flush: jest.fn() });
});

describe('UiFindInput', () => {
  const flushMock = jest.fn();

  const uiFind = 'uiFind';
  const ownInputValue = 'ownInputValue';
  const props = {
    uiFind: undefined,
    history: {
      replace: () => {},
    },
    location: {
      search: null,
    },
  };
  let wrapper;

  beforeAll(() => {
    debounceMock.mockImplementation((fn) => {
      function debounceFunction(...args) {
        fn(...args);
      }
      debounceFunction.flush = flushMock;
      return debounceFunction;
    });
  });

  beforeEach(() => {
    flushMock.mockReset();
    wrapper = shallow(<UiFindInput {...props} />);
  });

  describe('rendering', () => {
    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders props.uiFind when state.ownInputValue is `undefined`', () => {
      wrapper.setProps({ value: uiFind });
      expect(wrapper.find(UIInput).prop('value')).toBe(uiFind);
    });
  });
});
