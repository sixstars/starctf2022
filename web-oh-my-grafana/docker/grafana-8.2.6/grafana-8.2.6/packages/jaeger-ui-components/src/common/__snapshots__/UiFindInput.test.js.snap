// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiFindInput rendering renders as expected 1`] = `
<UIInput
  autosize={null}
  onChange={[Function]}
  placeholder="Find..."
  suffix={<React.Fragment />}
/>
`;
